kpm/lib/kpm/base_installer.rb

```diff
@@ -292,5 +315,16 @@ def extract_jruby_jar_version(jruby_jar)
       nil
     end
 
+    def get_plugin_name_from_file_path(file_path)
```

Comment:
Could you add a small unit test specifically for this, just to make sure we got the regex right? 😄 

Things to verify:

* `/Somewhere/xxx-foo/target/xxx-1.0.0.jar` -> `/var/tmp/mybundles/plugins/java/xxx`
* `/Somewhere/xxx-foo/target/xxx-foo-bar-1.0.0.jar` -> `/var/tmp/mybundles/plugins/java/xxx-foo-bar`
* `/Somewhere/xxx-foo/target/xxx-foo-1.0.0.jar` -> `/var/tmp/mybundles/plugins/java/xxx-foo`
* `/Somewhere/xxx-foo/target/xxx-foo-1.0.0-SNAPSHOT.jar` -> `/var/tmp/mybundles/plugins/java/xxx-foo`
* `/Somewhere/xxx-foo/target/xxx-foo-1.0.jar` -> `/var/tmp/mybundles/plugins/java/xxx-foo`
* `/Somewhere/xxx-foo/target/xxx-foo-1.jar` -> `/var/tmp/mybundles/plugins/java/xxx-foo`